astropy/utils/data.py

```diff
@@ -1019,90 +1019,103 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
     else:
         url_key = remote_url
 
-    try:
-        if cache:
-            # We don't need to acquire the lock here, since we are only reading
-            with _open_shelve(urlmapfn, True) as url2hash:
-                if url_key in url2hash:
-                    return url2hash[url_key]
+    url_key_list = url_key.split(',')
```

Comment:
I now also see that a list of strings for the default value is taken as "it has to be one of those". This would have been more logically done only if the input was a `set`, but I guess that is too late now... Anyway, probably best to stick with splitting on `,`.